core/src/main/scala/org/apache/spark/io/CompressionCodec.scala

```diff
@@ -170,9 +171,14 @@ class LZ4CompressionCodec(conf: SparkConf) extends CompressionCodec {
  */
 @DeveloperApi
 class LZFCompressionCodec(conf: SparkConf) extends CompressionCodec {
+  private val parallelCompression = conf.get(IO_COMPRESSION_LZF_PARALLEL)
 
   override def compressedOutputStream(s: OutputStream): OutputStream = {
-    new LZFOutputStream(s).setFinishBlockOnFlush(true)
+    if (parallelCompression) {
+      new PLZFOutputStream(s)
```

Comment:
This is creating a threadpool per `compressedOutputStream` - which can end up being quite expensive (num thread is num processors + some 'interesting' logic which tries to modulate it) .

Did you get a chance to try this on some nontrivial jobs ? Very curious about the experience.
Given this is turned off by default, dont see any concerns with the change itself though ! Would be a good way to understand the impact.